src/plasmapy/particles/particle_class.py

```diff
@@ -1852,6 +1854,13 @@ def ionize(self, n: int = 1, inplace: bool = False) -> Particle | None:
                 f"Cannot ionize {self.symbol} because it is not a "
                 f"neutral atom or ion."
             )
+        
+        if np.isinf(n):
```

Comment:
does this mean if n is infinity and inplace is the nucleus of the particle, then there is no issue, and nothing will be returned? Also does this force the particle to return its nucleus if n is infinity and inplace does not contain the particle's nucleus?